dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -189,31 +314,60 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
           Future.successful(value)
         case None =>
           Future.failed(
-            new RuntimeException(
+            new IllegalArgumentException(
               s"No event saved with nonce ${nonce.hex} $outcome"))
       }
 
-      eventOutcomeOpt <- eventOutcomeDAO.read((nonce, outcome))
+      eventOutcomeOpt <- eventOutcomeDAO.read((nonce, outcome.outcomeString))
       eventOutcomeDb <- eventOutcomeOpt match {
         case Some(value) => Future.successful(value)
         case None =>
-          Future.failed(new RuntimeException(
-            s"No event outcome saved with nonce and message ${nonce.hex} $outcome"))
+          Future.failed(new IllegalArgumentException(
+            s"No event outcome saved with nonce and message ${nonce.hex} ${outcome.outcomeString}"))
       }
 
-      sig = eventDb.signingVersion match {
-        case Mock =>
-          val kVal = getKValue(rValDb, Mock)
-          require(kVal.schnorrNonce == rValDb.nonce,
+      sigVersion = eventDb.signingVersion
+
+      kVal = getKValue(rValDb, sigVersion)
+      _ = require(kVal.schnorrNonce == rValDb.nonce,
                   "The nonce from derived seed did not match database")
-          signingKey.schnorrSignWithNonce(eventOutcomeDb.hashedMessage.bytes,
-                                          kVal)
-      }
+
+      hashBytes = eventOutcomeDb.hashedMessage
+      sig = signingKey.schnorrSignWithNonce(hashBytes, kVal)
 
       updated = eventDb.copy(attestationOpt = Some(sig.sig))
       _ <- eventDAO.update(updated)
     } yield updated
   }
+
+  def signLargeRange(
+      eventDescriptorTLV: LargeRangeEventDescriptorV0TLV,
+      num: Long): Future[OracleEvent] = {
+
+    // Make this a vec so it is easier to add on
+    val signSigF =
+      eventDescriptorTLV.signedNonceOpt match {
+        case Some(nonce) =>
+          val signOutcome = LargeRangeSignAttestation(num >= 0)
```

Comment:
Hmm... this doesn't seem right. For instance the number `1` can be both `signed` and `unsigned`, just checking if it's `>= 0` doesn't work does it?

It seems if we are signing numbers, the oracle needs to pre-commit if we are doing signed numbers or unsigned numbers, and then we need to track that. We can't just tell that at signing time can we?

Reply:
> For instance the number 1 can be both signed and unsigned, just checking if it's >= 0 doesn't work does it?

This is correct, `LargeRangeSignAttestation` takes in a boolean so we check if we want to sign "+" or "-" based on the number. If it is positive or 0 it signs "+", it is negative it signs "-".

> It seems if we are signing numbers, the oracle needs to pre-commit if we are doing signed numbers or unsigned numbers, and then we need to track that. We can't just tell that at signing time can we?

This is correct, we are only signing if it is positive or negative if we precommit to signing it, this is saved in the `eventDescriptorTLV`.